### PR TITLE
Fix(MessageButtonsBar): Hide copy message button for files

### LINKED ISSUE #10829: Files without caption allow copying the message
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Share a file to a conversation
2. See that there's still a "Copy formatted message" button in the context menu

### Expected behaviour
The button should not be available without a caption

### Actual behaviour
The placeholder `{file}` is copied to the clipboard.

## Talk app

**Talk app version:** main, tested on sermo

### PR DESCRIPTION
**Pull request by DorraJaouad**
### ☑️ Resolves

* Fix #10829

### 🖼️ Screenshots / Screencasts

| 🏚️ Before | 🏡 After |
|------------|----------|
| image         | image     |

### 🚧 Tasks

- [ ] Code review

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [x] 🖌️ Design was reviewed, approved or inspired by the design team
- [x] ⛑️ Tests are included or not possible
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

**Review by ShGKme — APPROVED** [review]

File: src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue [hunk 1]
```diff
@@ -583,32 +596,24 @@ export default {
...
+			const times = {
+				'Later today': laterTodayTime,
+				Tomorrow: tomorrowTime,
+				'This weekend': thisWeekendTime,
+				'Next week': nextWeekTime,
+			}
+
+			return Object.entries(times).map(([key, moment]) => {
+				return {
+					key: key.toLowerCase().replace(' ', '_'),
```
Comment (nickvergessen) [inline src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:608]: this does not match the previous keys? Would it be important?

File: src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue [hunk 2]
```diff
@@ -583,32 +596,24 @@ export default {
...
+					timestamp: this.getTimestamp(moment),
+					label: t('spreed', '{readableTime} – {timeLocale}', {
+						readableTime: key,
+						timeLocale: moment?.format('LT'),
+					}),
+					ariaLabel: t('spreed', `Set reminder for ${key.toLowerCase()}`),
```
Comment (nickvergessen) [inline src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:614]: This does not work for translations. Please revert to old pattern with fixed strings.

See https://docs.nextcloud.com/server/latest/developer_manual/basics/front-end/l10n.html#important-notes for more information

File: src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue (hunk 1 above)
Comment (DorraJaouad) [reply src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:608]: It is only used as key in loop and not elsewhere

File: src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue (hunk 2 above)
Comment (DorraJaouad) [reply src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:614]: The aria label is translatable but I am not sure if the label is translatable. I think that's where the translation broke ? 
image

Anyway, for such cases, how to translate strings with variable part that should be translated ?

Comment (nickvergessen) [reply src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:614]: > Anyway, for such cases, how to translate strings with variable part that should be translated ?

This is called string concatenation and does not work in all languages. Just don't use variabled strings. Go back to the predefined array that was here. (It's also unrelated to "Hide copy message button")

Comment (DorraJaouad) [reply src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue]: Yes, it was just an attempt to reduce the redundancy

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**